### PR TITLE
Migrate effective dart

### PR DESCRIPTION
**Pull request by munificent**
This forks the Markdown and code for "Effective Dart" and migrates the result to null safety.

I split the work into separate commits so that the mechanical changes aren't mixed with the real ones. The only substantive changes are in the last commit and just migrate the existing excerpts (and prose where needed) to make sense with null safety.

New guidelines and other clean up will be in subsequent PRs. :)

### CODE REVIEW

File: tool/dartformat.sh
Comment (kwalrath) [inline tool/dartformat.sh:47]: I think you might have to move up the entries you uncommented, so they're seen as part of the `dart format` command and not separate commands. At least, that's my guess as to why the build broke.

**Review by kwalrath — COMMENTED** [review]
LGTM once the build is fixed! Which engineer can we get to take a closer look at the code/style changes?

File: tool/dartformat.sh
Comment (munificent) [reply tool/dartformat.sh]: Done.

File: null_safety_examples/misc/lib/effective_dart/usage_bad.dart
Comment (natebosch) [inline null_safety_examples/misc/lib/effective_dart/usage_bad.dart:212]: This looks like it was a bug before. I think the intention was probably

```dart
Treasure? openChest(Chest chest, Point where) => _opened.containsKey(chest)
    ? null
    : _opened[chest] = (Treasure(where)..addAll(chest.contents));
```

Right?

**Review by natebosch — APPROVED** [review]

**Comment by munificent** [comment]
@natebosch reviewed the code now, but it still needs approval from someone who can submit to the site.

**Comment by kwalrath** [comment]
> @natebosch reviewed the code now, but it still needs approval from someone who can submit to the site.

I am that someone!

**Review by kwalrath — APPROVED** [review]

**Pull request merged**
